### PR TITLE
Refactor and clarify doc for load_results

### LINKED ISSUE #582: [doc] load_results loads all files matching an undocumented pattern and not just a single file as stated in the documentation
**Describe the bug**

load_results loads all results ending with 'monitor.csv' (and json files) and not just a single file as the documentation states.

**Code example**
Erroneous comment [here](https://github.com/hill-a/stable-baselines/blob/e315ebecbfd26225522a2e63a6bbd440c1df7f01/stable_baselines/bench/monitor.py#L163)

**System Info**
N/A

### PR DESCRIPTION
**Pull request by jbulow**
## Description
Updated doc string for load_results to match source code behavior. 
Remove duplicated code not using the correct Monitor.EXT constant for finding Monitor csv files.

## Motivation and Context
closes #582

- [ ] I have raised an issue to propose this change ([required](https://github.com/hill-a/stable-baselines/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x ] Documentation (update in the documentation)

## Checklist:

- [x ] I've read the [CONTRIBUTION](https://github.com/hill-a/stable-baselines/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the [changelog](https://github.com/hill-a/stable-baselines/blob/master/docs/misc/changelog.rst) accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.
- [ ] I have ensured `pytest` and `pytype` both pass.
pytests runs with error in unmodified main branch. pytype only available on old python version (3.5)

### CODE REVIEW

File: docs/misc/changelog.rst [hunk 1]
```diff
@@ -46,6 +46,7 @@ Others:
 - Add upper bound for Tensorflow version (<2.0.0).
 - Refactored test to remove duplicated code
 - Add pull request template
+- Replaced redundant code in load_results
```
Comment (araffin) [inline docs/misc/changelog.rst]: Please put your nickname too (and at the bottom).
Entries without are from maintainers.

File: tests/test_monitor.py [hunk 2]
```diff
@@ -34,3 +35,47 @@ def test_monitor():
...
     file_handler.close()
     os.remove(mon_file)
+
+def test_monitor_load_results(tmp_path):
+    """
+    test load_results on log files produced by the monitor wrapper
+    """
+    env1 = gym.make("CartPole-v1")
+    env1.seed(0)
+    mon_file1 = os.fspath(tmp_path / "stable_baselines-test-{}.monitor.csv".format(uuid.uuid4()))
```
Comment (araffin) [inline tests/test_monitor.py]: `os.fspath` is not available in python 3.5... (is this why you did not use `os.path.join` ?)

Comment (araffin) [inline tests/test_monitor.py]:
```suggestion
    monitor_file1 = os.fspath(tmp_path / "stable_baselines-test-{}.monitor.csv".format(uuid.uuid4()))
```

File: tests/test_monitor.py [hunk 3]
```diff
@@ -34,3 +35,47 @@ def test_monitor():
...
+    assert len(mon_files) == 1
+    assert mon_file1 in mon_files
+
+    menv1.reset()
+    for _ in range(1000):
+        _, _, done, _ = menv1.step(0)
+        if done:
+            menv1.reset()
+
+    r1s = load_results(os.fspath(tmp_path)).size
```
Comment (araffin) [inline tests/test_monitor.py]:
```suggestion
    results_size1 = load_results(os.fspath(tmp_path)).size
```

File: tests/test_monitor.py [hunk 4]
```diff
@@ -34,3 +35,47 @@ def test_monitor():
...
+    mon_file1 = os.fspath(tmp_path / "stable_baselines-test-{}.monitor.csv".format(uuid.uuid4()))
+    menv1 = Monitor(env1, mon_file1)
+
+    mon_files = get_monitor_files(tmp_path)
+    assert len(mon_files) == 1
+    assert mon_file1 in mon_files
+
+    menv1.reset()
+    for _ in range(1000):
+        _, _, done, _ = menv1.step(0)
```
Comment (araffin) [inline tests/test_monitor.py]: even though it works here, it is usually a better practice to do:
```python
action = env.action_space.sample()
```

File: tests/test_monitor.py (hunk 2 above)
Comment (jbulow) [reply tests/test_monitor.py]: Ah, my bad. The underlying problem is that load_results assumes the path is a string object by using `endswith` on it. I will find another solution.

File: tests/test_monitor.py (hunk 4 above)
Comment (jbulow) [reply tests/test_monitor.py]: ok. I will update. I just copied that code from the existing test to not deviate too much.

File: tests/test_monitor.py [hunk 5]
```diff
@@ -34,3 +35,47 @@ def test_monitor():
...
+
+    monitor_env2.reset()
+    for _ in range(1000):
+        _, _, done, _ = monitor_env2.step(monitor_env2.action_space.sample())
+        if done:
+            monitor_env2.reset()
+
+    results_size2 = load_results(os.path.join(tmp_path)).size
+
+    assert results_size2 > results_size1
```
Comment (araffin) [inline tests/test_monitor.py]: why is that?

File: tests/test_monitor.py [hunk 6]
```diff
@@ -34,3 +35,47 @@ def test_monitor():
...
+    assert monitor_file1 in monitor_files
+
+    monitor_env1.reset()
+    for _ in range(1000):
+        _, _, done, _ = monitor_env1.step(monitor_env1.action_space.sample())
+        if done:
+            monitor_env1.reset()
+
+    results_size1 = load_results(os.path.join(tmp_path)).size
+    assert results_size1 > 0
```
Comment (araffin) [inline tests/test_monitor.py]: you could predict that size by counting the number of episodes, no?

File: tests/test_monitor.py (hunk 4 above)
Comment (araffin) [reply tests/test_monitor.py]: It seems that the test comes from the original baselines and was apparently not updated...

File: tests/test_monitor.py (hunk 5 above)
Comment (jbulow) [reply tests/test_monitor.py]: The number of entries in the log file is less predictable when sampling from the action space, I guess.

It is not longer twice the size. results_size1 is 168 and results_size2 is 332 when I run and the numbers change between runs. It's fragile to depend too much random behavior in a test.

File: tests/test_monitor.py (hunk 6 above)
Comment (jbulow) [reply tests/test_monitor.py]: See above. The numbers differ between test runs.

Comment (jbulow) [reply tests/test_monitor.py]: Should I go back to using fixed action of 0 instead of sampling to get more predictable test runs?

Comment (araffin) [reply tests/test_monitor.py]: >The numbers differ between test runs.

that's normal but if you count the number of episode (looking the done signal), you can normally know in advance the size of the monitor file (cf evaluation helper: https://stable-baselines.readthedocs.io/en/master/_modules/stable_baselines/common/evaluation.html#evaluate_policy)

Comment (jbulow) [reply tests/test_monitor.py]: Missed your point with counting episodes at first. Now implemented.

File: tests/test_monitor.py (hunk 5 above)
Comment (jbulow) [reply tests/test_monitor.py]: Counting episodes now.

**Pull request merged**